### ISSUE TITLE
Pass SVG shape to startSelection handler

### ISSUE THREAD

**Issue by Vidyadayini** [annotorious/annotorious#197]
In order to add dynamic custom styling to an annotation being drawn, we need an option for the formatter function to be called on something like startSelection. Right now, the formatter works after the annotation is created or updated but not during creation.

```
anno.on('startSelection', function(point) {
  console.log(point.x, point.y)
});
```

A workaround I have is by actually updating the DOM on startSelection but we might not want to do that
```
 this.annotorious.on('startSelection', () => {
  const elem = document.querySelector('svg.a9s-annotationlayer .a9s-selection .a9s-inner');
  elem.style.stroke = my_color
});
```

**Comment by rsimon** [comment]
Formatter functions are for applying conditional styles, depending on the bodies stored in the annotation. (E.g. make an annotation red if it contains the tag "red", and blue if it contains the tag "blue".) When the user starts drawing a new selection, the annotation will be empty. There will be no information for the formatter to style on. 

If you want to apply a fixed custom style to all selections: that's much easier. You can simply do it with CSS.  New selection shapes have a `a9s-selection` class. CSS example below! (See also docs [here](https://recogito.github.io/guides/customizing-styles/).)

```css
svg.a9s-annotationlayer .a9s-selection .a9s-outer {
  // Your custom styles for outer selection SVG line
}

svg.a9s-annotationlayer .a9s-selection .a9s-inner {
  // Your custom styles for inner selection SVG line
}
```

**Comment by Vidyadayini** [comment]
Yes. I have these fixed styles but on startSelection, when I am trying to draw a new annotation, I do not want the fixed style for stroke color to appear. I want it to be dynamic. The formatter somehow is not being called here.

**Comment by rsimon** [comment]
Maybe I don't understand your use case correctly. The fixed style of the __annotations__ is defined via:

```css
svg.a9s-annotationlayer .a9s-annotation .a9s-outer { } // and .a9s-inner
```

When you draw a __new shape__, the style is defined via

```css
svg.a9s-annotationlayer .a9s-selection .a9s-outer { } // and .a9s-inner
```

Note the difference `a9s-annotation` vs. `a9s-selection`. If you want a new selections to be e.g. blue in one application state, and then in some other context, you want a new selection to be red, you'd need to dynamically modify the stylesheet via JS. Or am I mis-interpreting your use case?

**Comment by Vidyadayini** [comment]
Ok. So I will need to do something like this

```
 const elem = document.querySelector('svg.a9s-annotationlayer .a9s-selection .a9s-inner');
  elem.style.stroke = my_color
```

Because even the color is dynamic and we don't know what color the user will pick.

**Comment by rsimon** [comment]
Ok, understood. So the default colors are a user configuration? 

Yes, your solution above should work, if you do that in the `startSelection` handler. Another approach would be to modify the stylesheet dynamically, using [insertRule](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleSheet/insertRule), every time the user changes the setting. (Also see [here](https://fjolt.com/article/javascript-css#inserting-a-style-into-a-stylesheet) for an example.)

**Comment by rsimon** [comment]
Maybe it makes sense to pass the SVG shape to the startSelection event handler in the future. Then this would be even easier. I'll put that in the list!

**Comment by rsimon** [comment]
Closing this issue due to lack of activity. Annotorious v3 is almost ready to go. (It's now [a separate branch in this repo](https://github.com/annotorious/annotorious/tree/3.x). It will have a very different approach to rendering & styling (which should actually cover this use case).

**Comment by vidya-talapady-percipient-ai** [comment]
One question. Is this fix also available in 2.7 ?

**Comment by rsimon** [comment]
No, I'm afraid not.
